tests/integration/backends/test_tpp.py

```diff
@@ -2825,29 +2825,78 @@ def test_is_in_queries_on_columns_with_nonstandard_collation(
         (
             "?opensafely_include_t1oo=false",
             [
-                (1, 2001),
-                (4, 2004),
+                (1, 1951),
+                (4, 1954),
+                (6, 1956),
+                (8, 1958),
             ],
         ),
         (
             "?opensafely_include_t1oo=true",
             [
-                (1, 2001),
-                (2, 2002),
-                (3, 2003),
-                (4, 2004),
+                (1, 1951),
+                (2, 1952),
+                (3, 1953),
+                (4, 1954),
+                (5, 1955),
+                (6, 1956),
+                (7, 1957),
+                (8, 1958),
             ],
         ),
     ],
 )
 def test_t1oo_patients_excluded_as_specified(mssql_database, suffix, expected):
     mssql_database.setup(
-        Patient(Patient_ID=1, DateOfBirth=date(2001, 1, 1)),
-        Patient(Patient_ID=2, DateOfBirth=date(2002, 1, 1)),
-        Patient(Patient_ID=3, DateOfBirth=date(2003, 1, 1)),
-        Patient(Patient_ID=4, DateOfBirth=date(2004, 1, 1)),
+        # Included: no opt-out and a current registration
+        Patient(Patient_ID=1, DateOfBirth=date(1951, 1, 1)),
+        RegistrationHistory(
+            Patient_ID=1, StartDate=date(2020, 1, 1), EndDate=date(9999, 12, 31)
+        ),
+        # Excluded: current registration but opted out
+        Patient(Patient_ID=2, DateOfBirth=date(1952, 1, 1)),
+        RegistrationHistory(
+            Patient_ID=1, StartDate=date(2020, 1, 1), EndDate=date(9999, 12, 31)
+        ),
         PatientsWithTypeOneDissent(Patient_ID=2),
-        PatientsWithTypeOneDissent(Patient_ID=3),
+        # Excluded: no opt-out, but has deregistered
+        Patient(Patient_ID=3, DateOfBirth=date(1953, 1, 1)),
+        RegistrationHistory(
+            Patient_ID=3, StartDate=date(2020, 1, 1), EndDate=date(2024, 7, 1)
+        ),
+        # Included: no opt-out, has deregistered, but death recorded
+        Patient(
+            Patient_ID=4, DateOfBirth=date(1954, 1, 1), DateOfDeath=date(2024, 6, 1)
+        ),
+        RegistrationHistory(
+            Patient_ID=4, StartDate=date(2020, 1, 1), EndDate=date(2024, 7, 1)
+        ),
+        # Excluded: death recorded, registered, but had opt out in place
+        Patient(
+            Patient_ID=5, DateOfBirth=date(1955, 1, 1), DateOfDeath=date(2024, 7, 1)
+        ),
+        RegistrationHistory(
+            Patient_ID=5, StartDate=date(2020, 1, 1), EndDate=date(9999, 12, 31)
+        ),
+        PatientsWithTypeOneDissent(Patient_ID=5),
+        # Included: no opt-out and a deregistration followed by a subsequent
+        # re-registration
```

Comment:
In this case, could a patient have opted out during their previous registration (or during the period where they were registered elsewhere)? Do we assume that if they're not opted out now, it's OK to allow access to their data from the earlier registration period?  (I think so, just asking)

Reply:
> could a patient have opted out during their previous registration (or during the period where they were registered elsewhere)?

Existing opt-outs should transfer in along with the rest of the patient's records.

> Do we assume that if they're not opted out now, it's OK to allow access to their data from the earlier registration period?

Opting out (and opting back in again) covers all time periods. So it's not like some periods of patient history are covered by the opt out and others aren't: either the entire history needs to be excluded, or it doesn't.